android/app/src/main/java/com/microsoft/codepush/react/CodePush.java

```diff
@@ -193,6 +183,12 @@ void initializeUpdateAfterRestart() {
         // the app was just freshly updated.
         mDidUpdate = false;
 
+        JSONObject packageMetadata = this.mUpdateManager.getCurrentPackage();
+        if (!isPackageBundleLatest(packageMetadata) && hasBinaryVersionChanged(packageMetadata)) {
+            CodePushUtils.log("Skipping initializeUpdateAfterRestart(), binary version is newer");
+            return;
+        }
```

Comment:
Should I move this inside `if (pendingUpdate != null) {` to avoid reading the file `getCurrentPackage` file every time.